types/react-dom/canary.d.ts

```diff
@@ -147,4 +147,6 @@ declare module "./client" {
     interface HydrationOptions {
         formState?: ReactFormState | null;
     }
+
+    function createRoot(container: Document, options?: RootOptions): Root;
```

Comment:
not sure how these things layer together. Does this essentially support `Document | Element` when using canary and experimental builds?

Reply:
Yeah good call. Overloads are annoying anyway so I go with the same [pattern we use in `react` types to augment unions](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65220/files#diff-32cfd8cb197872bcba371f5018185d2e75fa540b52cda2dd7d8ac12dcc021299R242-R257)